rfcs/0037-ckb2021/0037-ckb2021.md

```diff
@@ -1 +1,108 @@
-In review, see <https://github.com/nervosnetwork/rfcs/pull/242>
+---
+Number: "0037"
+Category: Informational
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-07-24
+---
+
+# CKB2021 Overview
+
+The blockchain CKB is introducing a hard fork which will be complete by October, 2021. CKB2021 Overview outlines what you need to know for this upgrade.
+
+This is the first hard fork planned when CKB launches. The idea of a hard fork is to update the rules that are set to the network. The changes are not backward compatible. CKB2021 refers to the new CKB edition after the hard fork, where the old edition before the fork is CKB2019.
+
+## What's in CKB2021
+
+CKB2021 brings many new features and bug fixes. As usual, these changes are proposed via RFCs. The appendix has a list of all the RFCs related to CKB2021.
+
+The upgrade is divided into three categories. 
+
+First, CKB VM gets a major upgrade. CKB2021 will bundle CKB VM v1, in addition to the v0 in CKB2019. Scripts will be executed on v1 unless users opt in to use v0 by setting the script hash type to `data`.  
+
+Second, CKB2021 adds a new field `extension` in the block. This is reserved for future upgrades such as flyclient.
+
+Lastly, there are a bunch of consensus patches to fix bugs and make the consensus rules reasonable.
+
+### CKB VM v1
+
+Since CKB2021, there will be multiple VM versions available. ﻿[RFC32] designs a mechanism to choose the CKB VM version. It uses the `hash_type` field in the Script structure.
```

Comment:
```suggestion
Since CKB2021, there will be multiple VM versions available. ﻿[RFC32] introduces a CKB VM version mechanism. It piggybacks on the `hash_type` field in the Script structure.
```